### ISSUE TITLE
query params are null instead of empty object by default in beta2

### ISSUE THREAD

**Issue by TimothyKrell** [rackt/react-router#1476]
In 1.0.0-beta2 `location.query` appears to be `null` by default. I would have expected it to be an empty object like `params` (and I believe it was in previous versions). This leads to defensive checking if it is not null and more verbose code (e.g. `if (location.query && location.query.myParam)` rather than `if (location.query.myParam)`.

Also, why was query removed from top level props? It is a little less convenient to have to access it under `location`. I liked it being at `this.props.query` like `this.props.params`.

**Comment by TimothyKrell** [comment]
Setting `query` to `null` also breaks convenient destructuring workarounds.

```js
const {
  location: { query = {} }
} = this.params;

// query === null
```
If query was `undefined` this could work (although it still wouldn't be as nice as an empty object like `params`.

**Comment by knowbody** [comment]
As @mjackson said it's resolved on master (see: #1581 )
